server/chat.ts

```diff
@@ -1099,9 +1100,16 @@ export class CommandContext extends MessageContext {
 					const groupName = Config.groups[Config.pmmodchat] && Config.groups[Config.pmmodchat].name || Config.pmmodchat;
 					throw new Chat.ErrorMessage(this.tr`On this server, you must be of rank ${groupName} or higher to PM users.`);
 				}
-				if (targetUser.settings.blockPMs &&
-					(targetUser.settings.blockPMs === true || !Users.globalAuth.atLeast(user, targetUser.settings.blockPMs)) &&
-					!user.can('lock') && targetUser.id !== user.id) {
+				const targetFriends = Chat.Friends.cache.get(targetUser.id);
```

Comment:
So when the cache expires your friends will be randomly unable to PM you and then once the DB query finishes they can PM you? That seems confusing.

Reply:
When it expires, we don't remove the data - we merely make a new DB request to make it current.

Reply:
Ah. Wouldn't this still prevent the sending of PMs until that first request goes through?

Reply:
It's set up so that when you accept a request, or otherwise add a friend, the cache is updated automatically. If that hasn't resolved by the time you type out and send a PM, there's likely another problem.

Reply:
OK

Reply:
I just edited it to await the update of the cache for each user. Should be better.